test/Parse/ConditionalCompilation/basicParseErrors.swift

```diff
@@ -138,3 +138,9 @@ fn_k()
 func undefinedFunc() // ignored.
 #endif
 undefinedFunc() // expected-error {{use of unresolved identifier 'undefinedFunc'}}
+
+#if FOO
+#else if BAR
+// expected-error@-1 {{unexpected 'if' keyword following '#else' conditional compilation directive; did you mean '#elseif?'}}
```

Comment:
I think this is supposed to be just `expected-error`. Also, you can also check whether the fix-it is correctly applied by using this syntax:

```
{{column_start-column_end=<text>}}
```

so for example:

```
// expected-error {{unexpected 'if' keyword following '#else' conditional compilation directive; did you mean '#elseif?'}}{{0-9=#elseif}}
```

(wrote this without running, so you might have to tweak the range).

Reply:
`expected-error@-1` is correct; it says the error should appear on the line before the comment. You're correct that this should test the fix-it, though. Columns are 1-based (this confuses me all the time), so I think the range is 1-9.

Reply:
Sorry, I meant we could use `expected-error` and put it on the same line as `#else if BAR`? I suppose it's fine otherwise because the line would become even longer 😅

Reply:
Good catch, thanks!